libvast/vast/logger.hpp

```diff
@@ -122,50 +102,66 @@ auto id_or_name(T&& x) {
 
 #  endif // VAST_LOG_LEVEL > VAST_LOG_LEVEL_TRACE
 
-#  if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_ERROR
-#    define VAST_ERROR(...)                                                    \
-      VAST_LOG_COMPONENT(VAST_LOG_LEVEL_ERROR, __VA_ARGS__)
-#    define VAST_ERROR_ANON(...) VAST_LOG(VAST_LOG_LEVEL_ERROR, __VA_ARGS__)
-#  else
-#    define VAST_ERROR(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#    define VAST_ERROR_ANON(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#  endif
-
-#  if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_WARNING
-#    define VAST_WARNING(...)                                                  \
-      VAST_LOG_COMPONENT(VAST_LOG_LEVEL_WARNING, __VA_ARGS__)
-#    define VAST_WARNING_ANON(...) VAST_LOG(VAST_LOG_LEVEL_WARNING, __VA_ARGS__)
-#  else
-#    define VAST_WARNING(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#    define VAST_WARNING_ANON(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#  endif
-
-#  if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_INFO
-#    define VAST_INFO(...) VAST_LOG_COMPONENT(VAST_LOG_LEVEL_INFO, __VA_ARGS__)
-#    define VAST_INFO_ANON(...) VAST_LOG(VAST_LOG_LEVEL_INFO, __VA_ARGS__)
-#  else
-#    define VAST_INFO(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#    define VAST_INFO_ANON(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#  endif
-
-#  if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_VERBOSE
-#    define VAST_VERBOSE(...)                                                  \
-      VAST_LOG_COMPONENT(VAST_LOG_LEVEL_VERBOSE, __VA_ARGS__)
-#    define VAST_VERBOSE_ANON(...) VAST_LOG(VAST_LOG_LEVEL_VERBOSE, __VA_ARGS__)
-#  else
-#    define VAST_VERBOSE(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#    define VAST_VERBOSE_ANON(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#  endif
-
-#  if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_DEBUG
-#    define VAST_DEBUG(...)                                                    \
-      VAST_LOG_COMPONENT(VAST_LOG_LEVEL_DEBUG, __VA_ARGS__)
-#    define VAST_DEBUG_ANON(...) VAST_LOG(VAST_LOG_LEVEL_DEBUG, __VA_ARGS__)
-#  else
-#    define VAST_DEBUG(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#    define VAST_DEBUG_ANON(...) VAST_DISCARD_ARGS(__VA_ARGS__)
-#  endif
-
+#  define VAST_ERROR(c, ...)                                                   \
```

Comment:
If these are going to stay, we should switch the defaults so that `VAST_ERROR()` takes a format string and the old version is called `VAST_ERROR_LEGACY()`.

But if it turns out that we can perform the automated conversion with little effort this point becomes moot, so this doesn't need to be addressed immediately.

Reply:
Yes, once we agree on a strategy, we can make that.
The renaming to *_LEGACY' will be more easy , I guess :-) 
But lets first merge that in the epic branch, and then continue with the next action